### PR TITLE
[simd/v3i]: Implement more integer arithmetic instructions

### PR DESCRIPTION
**Pull request by haoyu-zc**
Tested by:
```
make v3i -j && bin/spectest.v3i -tk test/regress/simd/simd_i64x2_arith2.bin.wast
make v3i -j && bin/spectest.v3i -tk test/regress/simd/simd_i32x4_arith2.bin.wast
make v3i -j && bin/spectest.v3i -tk test/regress/simd/simd_i16x8_arith2.bin.wast
make v3i -j && bin/spectest.v3i -tk test/regress/simd/simd_i8x16_arith2.bin.wast
```

### CODE REVIEW

**Comment by haoyu-zc** [comment]
Rebased accordingly. But I'm not sure which utilities should be modified by private since I noticed you removed some `private`s in your latest commit. For methods such as `do_vv_v_x2`, I think they should be private bc other files may not use them, right?

**Comment by titzer** [comment]
Functions at the top level of a file (not in a class or a component) are private by default.

File: src/engine/V3Eval.v3 [hunk 1]
```diff
@@ -441,108 +498,259 @@ component V3Eval {
 			_ => return val;
 		}
 	}
-}
+	// Private (i.e. file-scoped) utilities.
```
Comment (titzer) [inline src/engine/V3Eval.v3:501]: Since these are outside the component, they don't need to be indented another level.

Comment (haoyu-zc) [reply src/engine/V3Eval.v3]: Oh, I got it. Didn't notice they were moved outside of the component. Pushed a new change.

**Review by titzer — APPROVED** [review]

**Pull request merged**